ingestion/src/metadata/ingestion/source/dashboard/superset/mixin.py

```diff
@@ -146,6 +152,58 @@ def _get_charts_of_dashboard(
             )
         return []
 
+    def parse_lineage_from_dataset_sql(
+        self, chart_json: FetchChart
+    ) -> list[tuple[FetchChart, dict[str, list[str]]]]:
+        # Every SQL query in tables is a SQL statement SELECTING data.
+        # To get lineage we 'simulate' INSERT INTO query into dummy table.
+        result = []
+
+        parser = LineageParser(f"INSERT INTO dummy_table {chart_json.sql}")
+        for table in parser.source_tables:
+            table_name = table.raw_name
+            table_schema = (
+                table.schema.raw_name
+                if table.schema.raw_name != table.schema.unknown
+                else chart_json.table_schema
+            )
+
+            column_mapping: dict[str, list[str]] = {}
+
+            for c in parser.column_lineage:
+                if "Table" in str(type(c[0].parent)) and "Table" in str(
+                    type(c[-1].parent)
+                ):
+                    if c[0].parent.schema.raw_name == table.schema.raw_name and c[0].parent.raw_name == table.raw_name:
+                        from_column_name = c[0].raw_name
+                        to_column_name = c[-1].raw_name
+
+                        if from_column_name != "*" and to_column_name != "*":
+                            if column_mapping.get(to_column_name):
+                                column_mapping[to_column_name].append(from_column_name)
+                            else:
+                                column_mapping[to_column_name] = [from_column_name]
+
+                        if from_column_name == "*" and to_column_name == "*":
+                            for col_name in self._get_columns_list_for_lineage(chart_json):
+                                if column_mapping.get(col_name):
+                                    column_mapping[col_name].append(col_name)
+                                else:
+                                    column_mapping[col_name] = [col_name]
```

Comment:
@mgorsk1 would you be able to break this down into individual functions where possible so we don't have a deeply nested code ?

Reply:
done :)